src/AstSemanticChecker.cpp

```diff
@@ -430,11 +430,35 @@ void AstSemanticChecker::checkLiteral(
         }
     }
 }
+/**
+ * agg1, agg2 are clauses which contain no head, and consist of a single literal
+ * that contains an aggregate.
+ * agg1 is dependent on agg2 if agg1 contains a variable which is grounded by agg2, and not by agg1.
+ */
+bool AstSemanticChecker::isDependent(const AstClause& agg1, const AstClause& agg2) {
+    auto grounded1 = getGroundedTerms(agg1);
+    auto grounded2 = getGroundedTerms(agg2);
+    bool dependent = false;
+    visitDepthFirst(agg1, [&](const AstVariable& var) {
+        const AstVariable* var2 = nullptr;
```

Comment:
Use more descriptive names, e.g. `searchVar` and `matchingVarPtr`. var and var2 aren't too terrible in this context, but it's a little confusing that they are not the same type.